src/js/stores/PluginStore.js

```diff
@@ -1,78 +1,121 @@
 import {EventEmitter} from "events";
 
 import AppDispatcher from "../AppDispatcher";
-import PluginDispatcher from "../plugin/external/PluginDispatcher";
+import Util from "../helpers/Util";
 import DialogActions from "../actions/DialogActions";
 import DialogSeverity from "../constants/DialogSeverity";
 import PluginActions from "../actions/PluginActions";
 import PluginEvents from "../events/PluginEvents";
+import PluginModules from "../constants/PluginModules";
+import States from "../constants/States";
+import pluginScheme from "../stores/schemes/pluginScheme";
 
-import Util from "../helpers/Util";
+var plugins = [];
+
+function addPlugin(data) {
+  if (data.id == null && !Util.isObject(data.info)
+    && getPluginById(data.id) != null) {
+    return;
+  }
 
-var pluginsMetaData = [];
-var pluginsToLoad = [];
-var pluginsStarted = [];
-var pluginsErrored = [];
+  plugins.push(Util.extendObject(pluginScheme, data.info, {id: data.id}));
+}
 
-var bootstrapComplete = false;
+function getPluginById(id) {
+  return plugins.find((plugin) => plugin.id === id);
```

Comment:
You can get rid of the parantheses of the 1st argument.
`return plugins.find(plugin => plugin.id === id);`